src/makeCalculator.js

```diff
@@ -38,6 +38,37 @@
  */
 function makeCalculator() {
   // write code here
+  return {
+    result: 0,
+
+    add(number, calculator) {
+      calculator.result += number;
+    },
+
+    subtract(number, calculator) {
+      calculator.result -= number;
+    },
+
+    multiply(number, calculator) {
+      calculator.result *= number;
+    },
+
+    divide(number, calculator) {
+      calculator.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      callback(number, this);
```

Comment:
you can do that in a cleaner way. the reason you need `this` is because you need to pass the context. you can do something like `callback.call(this, number)` and you no longer need `calculator` as a parameter of each operation function, simply use this. 

it's just a suggestion

Reply:
thank you